apps/application-system/api/src/app/modules/application/application.module.ts

```diff
@@ -1,25 +1,34 @@
-import { Module } from '@nestjs/common'
-import { BullModule } from '@nestjs/bull'
+import { DynamicModule, Module } from '@nestjs/common'
+import { BullModule as NestBullModule } from '@nestjs/bull'
 import { SequelizeModule } from '@nestjs/sequelize'
+import { FileStorageModule } from '@island.is/file-storage'
+
 import { Application } from './application.model'
 import { ApplicationController } from './application.controller'
 import { ApplicationService } from './application.service'
-import { FileStorageModule } from '@island.is/file-storage'
 import { UploadProcessor } from './upload.processor'
 
+let BullModule: DynamicModule
+
+if (process.env.INIT_SCHEMA === 'true') {
+  BullModule = NestBullModule.registerQueueAsync()
```

Comment:
Probably needs to have the same name (upload). I think it's better to put this kind of stuff inside the async useFactory (possibly inline with the @Module declaration like it was):

`
useFactory: () => (inMemory ? {} : { redis: ... }),
`

Reply:
I don't know why, but with this syntax, the process still keeps running and doesn't stop

Reply:
Ah ... Maybe because of the name parameter, triggering the consumer to start waiting for upload jobs.

Reply:
Yes adding the name, trigger the upload job.

Should I keep it this way?

Reply:
Sure, until we find a better way...